src/pages/AccountManagementPage.tsx

```diff
@@ -0,0 +1,227 @@
+      })
+  }
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      resetPassword(password)
+    }
+  }
+
+  return (
+    <Box sx={{ p: 10 }}>
+      <Container maxWidth="md">
+        <Paper
+          sx={{
+            p: 10,
+            m: 5,
+          }}
+        >
+          <Stack spacing={3} alignItems={'center'}>
+            <Typography variant="h3">Reset Password</Typography>
+            <TextField
+              fullWidth
+              label="New Password"
+              variant="standard"
+              onChange={(e) => setPassword(e.target.value)}
+              onKeyDown={(e) => handleKeyPress(e)}
+            ></TextField>
+            <Box>
+              <Typography
+                fontSize={'18px'}
+                color={'red'}
+                visibility={hasInvalidCredentials ? 'visible' : 'hidden'}
+              >
+                Password is too weak. Please use more than 4 characters.
+              </Typography>
+            </Box>
+            <StylishButton
+              variant="outlined"
+              onClick={() => resetPassword(password)}
+            >
+              Submit
+            </StylishButton>
+          </Stack>
+        </Paper>
+      </Container>
+    </Box>
+  )
+}
+
+export default function AccountManagementPage() {
+  const [mode, setMode] = useState<string>('')
+  const [hasValidOob, setHasValidOob] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+
+  const [searchParams] = useSearchParams()
+  const user = useSelector((state: RootState) => state.user)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const urlMode = searchParams.get('mode')
+    const oob = searchParams.get('oobCode')
+
+    if (urlMode === 'resetPassword') {
+      fetch(buildPath('/checkResetPassword'), {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ oobCode: oob }),
+      })
+        .then((res: Response) => {
+          if (res.ok) {
+            setHasValidOob(true)
+            setIsLoading(false)
+            return res.json()
+          }
+
+          throw Error(res.statusText)
+        })
+        .catch(() => {
+          setIsLoading(false)
+          setHasValidOob(false)
+        })
+    } else {
+      fetch(buildPath('/doEmailVerification'), {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ oobCode: oob }),
+      })
+        .then((res: Response) => {
+          if (res.ok) {
+            setHasValidOob(true)
+            setIsLoading(false)
+            dispatch(verify())
+            return res.json()
+          }
+
+          throw Error(res.statusText)
+        })
+        .catch(() => {
+          setIsLoading(false)
+          setHasValidOob(false)
+        })
+    }
+
+    setMode(urlMode || '')
+  }, [])
+
+  if (isLoading) {
+    return <React.Fragment />
```

Comment:
If you end up creating a loading screen (i would say just an circle progress icon centered in page and rotating) for problem submit page, we can use it here and anywhere we have isLoading with empty react fragment.